pkg/cosign/keys.go

```diff
@@ -51,6 +60,74 @@ func GeneratePrivateKey() (*ecdsa.PrivateKey, error) {
 	return ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
 }
 
+func ImportKeyPair(keyPath string, pf PassFunc) (*Keys, error) {
+
+	kb, err := os.ReadFile(filepath.Clean(keyPath))
+	if err != nil {
+		return nil, err
+	}
+
+	p, _ := pem.Decode(kb)
+	if p == nil {
+		return nil, errors.New("invalid pem block")
+	}
+	if p.Type != RSAPrivateKeyPemType && p.Type != ECPrivateKeyPemType {
+		return nil, fmt.Errorf("unsupported pem type: %s", p.Type)
+	}
+
+	if p.Type == RSAPrivateKeyPemType {
+		pk, err := x509.ParsePKCS1PrivateKey(p.Bytes)
+		if err != nil {
+			return nil, fmt.Errorf("parsing error")
+		}
+		return MarshallKeyPair(Key{pk, pk.Public()}, pf)
+	} else {
```

Comment:
I wonder if this could be turned into an elif, and have an else play the role of 74-75.

Reply:
Made changes.